service/controller/v3/resource/legacy/resource.go

```diff
@@ -227,13 +226,13 @@ func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desir
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-	if len(list.Items) != 0 {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "cannot finish deletion of network due to existing pods")
-		finalizerskeptcontext.SetKept(ctx)
-		resourcecanceledcontext.SetCanceled(ctx)
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-
-		return nil, nil
+	for _, pod := range list.Items {
+		err := r.k8sClient.CoreV1().Pods(pod.GetNamespace()).Delete(pod.GetName(), &metav1.DeleteOptions{})
+		if apierrors.IsNotFound(err) {
+			continue
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		}
```

Comment:
IIUIC, then this is controversial. Why would `flannel-operator` delete pods created by `kvm-operator`? Previous implementation seemed ok.

Reply:
I am really confused now because this code deleted the flannel pods in the flannel namespace.

Reply:
Any chance it was coincidence with fine timing? Isn't the problem with flannel PODs that they sometimes get deleted and sometimes don't?